onmt/modules/StackedRNN.py

```diff
@@ -47,14 +47,18 @@ def __init__(self, num_layers, input_size, rnn_size, dropout):
             input_size = rnn_size
 
     def forward(self, input, hidden):
+        h_0 = hidden[0]
         h_1 = []
         for i, layer in enumerate(self.layers):
-            h_1_i = layer(input, hidden[i])
+            if len(h_0.size()) == 3:
```

Comment:
Please avoid magic numbers like this when possible. If you can put a comment indicate the type of hidden. Even better add `aeq` assertions (like in GlobalAttention) to make sure the sizes are as expected.